lib/hipchat/http.ex

```diff
@@ -26,14 +26,9 @@ defmodule Hipchat.HTTP do
   Builds headers
   """
   def set_headers(client) do
-    HashDict.new
-    |> authentication(client[:token])
-    |> HashDict.put(:"Accept", "application/json")
-    |> HashDict.put(:"content-type", "application/json")
-  end
-
-  defp authentication(hash, token) do
-    HashDict.put(hash, :"Authorization", "Bearer " <> token)
+    ["Authorization": "Bearer " <> (client[:token] != nil && client[:token] || ""),
```

Comment:
First off thanks for the PR.  
I am thinking something a little different here.
Since elixir 1.1 will deprecate Access syntax `opts[key]` we can use `Dict.get/3` instead.
For example,
``` elixir
["Authorization": "Bearer " <> (to_string(Dict.get(client, :token))), ...
```
`to_string/1` will take care of the defensive needs to check for existence of `:token`

Brad